src/features/posts.ts

```diff
@@ -0,0 +1,45 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { Post } from '../types/Post';
+import { User } from '../types/User';
+import { getUserPosts } from '../api/posts';
+
+type PostsState = {
+  posts: Post[];
+  loaded: boolean;
+  hasError: boolean;
+};
+
+const initialState: PostsState = {
+  posts: [],
+  loaded: false,
+  hasError: false,
+};
+
+export const getPosts = createAsyncThunk<Post[], User['id']>(
+  'posts/fetch',
+  userId => getUserPosts(userId),
+);
+
+export const postsSlice = createSlice({
+  name: 'posts',
+  initialState,
+  reducers: {},
+  extraReducers: builder => {
+    builder
+      .addCase(getPosts.fulfilled, (state, action) => {
+        return {
+          ...state,
+          posts: action.payload,
+        };
+      })
+      .addCase(getPosts.pending, state => {
+        return { ...state, loaded: false };
+      })
+      .addCase(getPosts.rejected, state => {
+        return { ...state, hasError: true };
```

Comment:
In your rejected case, you're only setting hasError to true. It would be better if you also reset the posts array and the loaded state to their initial state. This way, in case of an error, your application won't display incorrect or stale data.